.claude/hooks/tests/test_git_header_statusline.py

```
@@ -209,14 +215,219 @@ def test_red_matrix_uncommitted_scenarios(
+        return Path("/tmp") / f"git-header-pr-{commit[:8]}"
+
+    def test_pr_cache_uses_cached_value_when_fresh(
+        self, make_temp_repo_with_remote, git_header_script, gh_stub
+    ):
+        """Cache should prevent repeated gh calls while within PR TTL."""
+
+        _script, call_log, env = gh_stub
+        repo_dir, branch_name, _ = make_temp_repo_with_remote()
+
+        git_dir = (
+            subprocess.check_output(["git", "rev-parse", "--git-common-dir"], cwd=repo_dir)
```

Ruff (0.14.8)

Starting a process with a partial executable path

(S607)